common/lib/dependabot/file_updaters/vendor_updater.rb

```diff
@@ -31,8 +31,10 @@ def updated_vendor_cache_files(base_directory:)
               encoding = Dependabot::DependencyFile::ContentEncoding::BASE64
               encoded_content = Base64.encode64(encoded_content) unless deleted
             end
+
+            clean_base_directory = Pathname.new(base_directory).basename
```

Comment:
Is `basename` the right thing here? Looks like it just grabs the last component `Pathname.new("dir/nested").basename` > `nested` 

Should it instead be something like?

```
project_root  = Pathname.new(File.expand_path(File.join(Dir.pwd, base_directory)))
filename = Pathname.new(path).expand_path.relative_path_from(project_root)

Reply:
Yeah great suggestion, I've improved the other relative path code as well